### DISCUSSION TITLE
Is the `digest()` method for an object used for creating the hash?

### DISCUSSION THREAD

**Discussion by billdenney** [ropensci/targets#787] [Help] [answered]
While working on the `nlmixr2targets` update, I am working to simplify some of the work done in the package and push appropriate work upstream into the `nlmixr2` package (and other, related packages).  I *think* that `targets` uses the `digest()` of the dependencies to determine if they have changed.  But, when I looked at https://books.ropensci.org/targets-design/data.html#metadata, I didn't specifically see that stated.  I see the words "hash of" used several times, there, but I don't explicitly see how the hash is created.

Sorry if I'm missing something in the docs, but I want to make sure that I get this right.  (I also glanced around the source, and see `digest()` used a lot, so I'm pretty confident, but I don't want to have many revisions to get it working.)

**Comment by wlandau** [comment, marked as answer]
Yup, targets uses digest to hash: https://github.com/ropensci/targets/blob/8f91e506c27900eb6af4527319e0b0eb5c180ffc/R/utils_digest.R

**Reply by billdenney** [reply]
Thanks!
